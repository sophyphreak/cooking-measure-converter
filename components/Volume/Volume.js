import React from 'react'
import TableRow from '../../containers/TableRow/TableRow'
import getPropsForTableRow from './getPropsForTableRow/getPropsForTableRow'

const Volume = ({ volumeProps }) => {
  const {
    metricDropdown,
    metricInput,
    imperialInput,
    imperialDropdown
  } = getPropsForTableRow(volumeProps)
  return (
    <TableRow
      metricDropdown={metricDropdown}
      metricInput={metricInput}
      imperialInput={imperialInput}
      imperialDropdown={imperialDropdown}
    />
  )
}

export default Volume
