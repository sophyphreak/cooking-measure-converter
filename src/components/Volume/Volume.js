import React from 'react';
import TableRow from '../TableRow/TableRow';

const Volume = ({ volumeProps }) => {
  const { volume, onImperialVolumeChange, onMetricVolumeChange } = volumeProps;
  const { imperialVolume, imperialUnit, metricVolume, metricUnit } = volume;
  return (
    <TableRow
      metricUnit={metricUnit}
      metricValue={metricVolume}
      onMetricChange={onMetricVolumeChange}
      imperialValue={imperialVolume}
      onImperialChange={onImperialVolumeChange}
      imperialUnit={imperialUnit}
    />
  );
};

export default Volume;
