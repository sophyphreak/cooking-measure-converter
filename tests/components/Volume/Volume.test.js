import React from 'react';
import { shallow } from 'enzyme';
import Volume from '../../../src/components/Volume/Volume';

let volume, onImperialVolumeChange, onMetricVolumeChange, volumeProps, wrapper;

beforeEach(() => {
  volume = {
    imperialVolume: '',
    imperialUnit: 'Tbs',
    metricVolume: '',
    metricUnit: 'ml'
  };
  onImperialVolumeChange = jest.fn();
  onMetricVolumeChange = jest.fn();
  volumeProps = {
    volume,
    onImperialVolumeChange,
    onMetricVolumeChange
  };
  wrapper = shallow(<Volume volumeProps={volumeProps} />);
});

test('should render Volume correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
