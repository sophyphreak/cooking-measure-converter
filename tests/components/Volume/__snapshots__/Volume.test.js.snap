// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Volume correctly 1`] = `
<TableRow
  imperialUnit="Tbs"
  imperialValue=""
  metricUnit="ml"
  metricValue=""
  onImperialChange={[MockFunction]}
  onMetricChange={[MockFunction]}
/>
`;
